api/controllers/CampaignBotController.js

```diff
@@ -70,30 +67,33 @@ CampaignBotController.prototype.chatbot = function(req, res) {
       req.incoming_image_url);
   }
 
-  dbUsers.findOne({ '_id': req.user_id }, function (err, userDoc) {
-
-    if (err) {
-      return self.handleError(req, res, err);
-    }
+  this.loadUserAndSignup(req, res);
+}
 
+/** 
+ * Creates user and posts Signup to the Campaign.
+ * @param {object} req - Express request
+ * @param {object} res - Express response
+ */
+CampaignBotController.prototype.loadUserAndSignup = function(req, res) {
+  logger.debug('%s loadUserAndSignup', this.loggerPrefix(req));
+  const promise = dbUsers.findOne({ '_id': req.user_id }).exec();
```

Comment:
Also, it's recommended to use [`Model.findById()`](http://mongoosejs.com/docs/api.html#query_Query-findOne):

> Note: conditions is optional, and if conditions is null or undefined, mongoose will send an empty findOne command to MongoDB, which will return an arbitrary document. If you're querying by _id, use Model.findById() instead.